networkx/algorithms/tree/branchings.py

```diff
@@ -1083,6 +1080,7 @@ def is_root(G, u, edgekeys):
 
             edmonds_add_edge(B, B_edge_index, u, v, desired_edge[2], **dd)
             G[u][v][desired_edge[2]][candidate_attr] = True
+            nx._clear_cache(G)
```

Comment:
Done! Disabling the cache for `G` is a good idea here, since `G` is an implementation detail that is never exposed to the user, and I don't think there's benefit for it to actually use dispatching. Good catch!